### PR TITLE
chore: add combobox spec

### PR DESCRIPTION
**Pull request by radium-v**
# Description

Adds the specification for the `combobox` component.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: packages/web-components/fast-foundation/src/combobox/combobox.spec.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+## Overview
+
+As defined by the [W3C](https://w3c.github.io/aria-practices/#combobox):
+> A combobox is an input widget with an associated popup that enables users to select a value for the combobox from a collection of possible values. In some implementations, the popup presents allowed values, while in other implementations, the popup presents suggested values, and users may either select one of the suggestions or type a value. The popup may be a listbox, grid, tree, or dialog. Many implementations also include a third optional element -- a graphical Open button adjacent to the combobox, which indicates availability of the popup. Activating the Open button displays the popup if suggestions are available.
+
+### Use Cases
+
+- *A customer using the component on a web page.*
+On a web page a customer is providing their mailing address for an order form. The customer clicks the state `<fast-combobox>` and sees a list of options. The customer begins typing their state and the field is automatically populated with the closest matching state. Then, the customer uses the arrow keys to highlight their state and presses `enter` to commit the value to the combobox.
```
Comment (janechu) [inline packages/web-components/fast-foundation/src/combobox/combobox.spec.md:11]: I find the use of the word "state" slightly confusing from a non US perspective (some countries have counties, provinces, etc.) and also from the perspective of the "state" in programming terms, so here's a suggestion just for additional clarity.

```suggestion
On a web page a customer is providing their US mailing address for an order form. The customer clicks a `<fast-combobox>` to fill in their state (California, Ohio, etc.) and sees a list of options. The customer begins typing the state name and the field is automatically populated with the closest matching state. Then, the customer uses the arrow keys to highlight their state and presses `enter` to commit the value to the combobox.
```

File: packages/web-components/fast-foundation/src/combobox/combobox.spec.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+### Features
+
+- **Form association**: The `combobox` is [form-associated](../form-associated/form-associated-custom-element.spec.md), allowing its selected or user-provided value to be submitted with a form.
+- **Autocomplete and Autoselection**: Supports `inline`, `list`, and `both` values for `autocomplete`:
+  - `inline`: automatically fills the value of the first matching option while typing, and highlights the auto-inserted text. While a value is present, navigating through the list will automatically jump to partially-matching options.
+  - `list`: auto-filters the available options to only display those which start with the current value.
+  - `both`: automatically filters the list and fills the remaining text for the currently selected option.
```
Comment (janechu) [inline packages/web-components/fast-foundation/src/combobox/combobox.spec.md:19]:
```suggestion
  - `inline`: Automatically fills the value of the first matching option while typing, and highlights the auto-inserted text. While a value is present, navigating through the list will automatically jump to partially-matching options.
  - `list`: Auto-filters the available options to only display those which start with the current value.
  - `both`: Automatically filters the list and fills the remaining text for the currently selected option.
```

File: packages/web-components/fast-foundation/src/combobox/combobox.spec.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+*Properties*:
+
+- `options` - An array of all options.
+- `placeholder` - Sets the placeholder of the element, generally used to provide a hint to the user.
+- `role` - The role of the element.
+- `value` - Reflects the value of the first selected option. Setting the `value` property will update the `selected` state of the first option that matches the value, if any.
+
+*Events*:
+
+- `change` - emits when the `value` is changed via user interaction.
```
Comment (janechu) [inline packages/web-components/fast-foundation/src/combobox/combobox.spec.md:62]:
```suggestion
- `change` - Emits when the `value` is changed via user interaction.
```

File: packages/web-components/fast-foundation/src/combobox/combobox.spec.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+*Parts*:
+
+- `control` - the container that holds the `start`, `end`, `control`, and `indicator` slots.
+- `indicator` - the container that holds the `select-indicator` slot.
+- `select-indicator` - the default SVG icon present the `indicator` slot.
+- `listbox` - The container that represents the listbox popup.
+- `selected-value` - the container that holds the displayed text for the button.
+- `start` - the container that holds the `start` slot.
+- `end` - the container that holds the `end` slot.
```
Comment (janechu) [inline packages/web-components/fast-foundation/src/combobox/combobox.spec.md:80]:
```suggestion
- *default* - The list of options, either `<fast-option>` or elements with `role="option"`.
- `control` - Contains the selected value and indicator slots inside the control element.
  - `indicator` - Holds the glyph indicating that the combobox can be expanded. This slot is only available if the `control` slot is not filled.
- `start` - Used to display content like glyphs or icons inside the button, before the control.
- `end` - Used to display content like glyphs or icons inside the button, after the control.

*Parts*:

- `control` - The container that holds the `start`, `end`, `control`, and `indicator` slots.
- `indicator` - The container that holds the `select-indicator` slot.
- `select-indicator` - The default SVG icon present the `indicator` slot.
- `listbox` - The container that represents the listbox popup.
- `selected-value` - The container that holds the displayed text for the button.
- `start` - The container that holds the `start` slot.
- `end` - The container that holds the `end` slot.
```

File: packages/web-components/fast-foundation/src/combobox/combobox.spec.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+**Note**: While `combobox` is implemented in very similar ways as [`select`](../select/select.spec.md), the `value` attribute on `fast-option` and other `option`-like elements is ignored in favor of the `textContent`. This is done to avoid confusion between user-provided values and predefined options.
+
+### States
+
+- `open` -  This state is applied to the `<fast-combobox>` when the listbox is visible to the user.
+- `required` - The `<fast-combobox>` must have a user-provided value when the `required` attribute is set to `true`.
+- `valid` - The `<fast-combobox>` meets all its validation constraints, and is therefore considered to be valid.
+- `invalid` - The `<fast-combobox>` does not meet its validation constraints, and is therefore considered to be invalid.
+- `disabled` - when disabled, the value will not be changeable through user interaction.
```
Comment (janechu) [inline packages/web-components/fast-foundation/src/combobox/combobox.spec.md:141]:
```suggestion
- `disabled` - When disabled, the value will not be changeable through user interaction.
```

**Review by janechu — CHANGES_REQUESTED** [review]
Couple nits, I would also say do not mark this as a feat as only documentation has changed, this should be a chore.

File: packages/web-components/fast-foundation/src/combobox/combobox.spec.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+*Properties*:
+
+- `options` - An array of all options.
+- `placeholder` - Sets the placeholder of the element, generally used to provide a hint to the user.
+- `role` - The role of the element.
+- `value` - Reflects the value of the first selected option. Setting the `value` property will update the `selected` state of the first option that matches the value, if any.
+
+*Events*:
+
+- `change` - Emits when the `value` is changed via user interaction.
```
Comment (nicholasrice) [inline packages/web-components/fast-foundation/src/combobox/combobox.spec.md:62]: should it also emit an `input` event on input?

Comment (radium-v) [reply packages/web-components/fast-foundation/src/combobox/combobox.spec.md:62]: An `input` event is emitted from the internal input, but I wasn't sure if the text box's native events should be considered as part of the spec here. I can add it if that is the case.

**Review by janechu — APPROVED** [review]

File: packages/web-components/fast-foundation/src/combobox/combobox.spec.md (hunk 6 above)
Comment (EisenbergEffect) [reply packages/web-components/fast-foundation/src/combobox/combobox.spec.md:60]: I think it could be valuable to add it to the spec, if only for clarity.

**Review by EisenbergEffect — APPROVED** [review]

**Pull request merged**